xarray/tests/test_dataarray.py

```diff
@@ -4620,6 +4622,72 @@ def test_pad_reflect(self, mode, reflect_type):
         assert actual.shape == (7, 4, 9)
         assert_identical(actual, expected)
 
+    @requires_dask
+    @requires_numexpr
+    @pytest.mark.parametrize("parser", ["pandas", "python"])
+    @pytest.mark.parametrize("engine", ["python", "numexpr", None])
+    @pytest.mark.parametrize("backend", ["numpy", "dask"])
```

Comment:
I think we should skip only where necessary:
```suggestion
    @pytest.mark.parametrize("backend", ["numpy", pytest.param("dask", marks=[requires_dask])])
```
that way, the `all-but-dask` CI can check whether this works without having `dask` installed. The same for `numexpr`, but I'm not sure what to do with `engine=None`.

Reply:
Nice approach. Should be ok to let `engine=None` run without numexpr, pandas should try numexpr and fall back to python.

Reply:
Implemented.